### ISSUE TITLE
Crash with TaskGroupError

### ISSUE THREAD

**Issue by emanuelelaface** [kyuupichan/electrumx#542]
The latest release (1.7.2) crashes after couple of hours with this log

```
Jul 31 14:46:44 hook electrumx_server[4680]: ERROR:TaskGroup:task crashed: <Task finished coro=<PeerManager._send_headers_subscribe() done, defined at /opt/python/lib/python3.6/site-packages/electrumX-1.7.2-py3.6.egg/electrumx/server/peers.py:310> exception=BadPeerError('bad height 534,485 (ours: 534,540)',)>
Jul 31 14:46:44 hook electrumx_server[4680]: Traceback (most recent call last):
Jul 31 14:46:44 hook electrumx_server[4680]:   File "/opt/python/lib/python3.6/site-packages/aiorpcX-0.5.9-py3.6.egg/aiorpcx/curio.py", line 138, in _on_done
Jul 31 14:46:44 hook electrumx_server[4680]:     task.result()
Jul 31 14:46:44 hook electrumx_server[4680]:   File "/opt/python/lib/python3.6/site-packages/electrumX-1.7.2-py3.6.egg/electrumx/server/peers.py", line 323, in _send_headers_subscribe
Jul 31 14:46:44 hook electrumx_server[4680]:     raise BadPeerError(f'bad height {their_height:,d} '
Jul 31 14:46:44 hook electrumx_server[4680]: electrumx.server.peers.BadPeerError: bad height 534,485 (ours: 534,540)
Jul 31 14:46:44 hook electrumx_server[4680]: ERROR:PeerManager:[134.119.179.55] marking bad: (server.version returned bad result type str)
Jul 31 14:46:44 hook electrumx_server[4680]: INFO:PeerManager:failed to verify 134.119.179.55 via SSL at 134.119.179.55 in 0.6s
Jul 31 14:46:44 hook electrumx_server[4680]: INFO:PeerManager:verified e-4.claudioboxx.com via SSL at 134.119.179.67 in 1.0s
Jul 31 14:46:44 hook electrumx_server[4680]: ERROR:TaskGroup:task crashed: <Task finished coro=<PeerManager._monitor_peer() done, defined at /opt/python/lib/python3.6/site-packages/electrumX-1.7.2-py3.6.egg/electrumx/server/peers.py:185> exception=TaskGroupError([<Task finished coro=<PeerManager._send_headers_subscribe() done, defined at /opt/python/lib/python3.6/site-packages/electrumX-1.7.2-py3.6.egg/electrumx/server/peers.py:310> exception=BadPeerError('bad height 534,485 (ours: 534,540)',)>],)>
Jul 31 14:46:44 hook electrumx_server[4680]: Traceback (most recent call last):
Jul 31 14:46:44 hook electrumx_server[4680]:   File "/opt/python/lib/python3.6/site-packages/aiorpcX-0.5.9-py3.6.egg/aiorpcx/curio.py", line 138, in _on_done
Jul 31 14:46:44 hook electrumx_server[4680]:     task.result()
Jul 31 14:46:44 hook electrumx_server[4680]:   File "/opt/python/lib/python3.6/site-packages/electrumX-1.7.2-py3.6.egg/electrumx/server/peers.py", line 188, in _monitor_peer
Jul 31 14:46:44 hook electrumx_server[4680]:     if await self._should_drop_peer(peer):
Jul 31 14:46:44 hook electrumx_server[4680]:   File "/opt/python/lib/python3.6/site-packages/electrumX-1.7.2-py3.6.egg/electrumx/server/peers.py", line 227, in _should_drop_peer
Jul 31 14:46:44 hook electrumx_server[4680]:     await self._verify_peer(session, peer)
Jul 31 14:46:44 hook electrumx_server[4680]:   File "/opt/python/lib/python3.6/site-packages/electrumX-1.7.2-py3.6.egg/electrumx/server/peers.py", line 308, in _verify_peer
Jul 31 14:46:44 hook electrumx_server[4680]:     timeout))
Jul 31 14:46:44 hook electrumx_server[4680]:   File "/opt/python/lib/python3.6/site-packages/aiorpcX-0.5.9-py3.6.egg/aiorpcx/curio.py", line 254, in __aexit__
Jul 31 14:46:44 hook electrumx_server[4680]:     await self.join(wait=self._wait)
Jul 31 14:46:44 hook electrumx_server[4680]:   File "/opt/python/lib/python3.6/site-packages/aiorpcX-0.5.9-py3.6.egg/aiorpcx/curio.py", line 228, in join
Jul 31 14:46:44 hook electrumx_server[4680]:     raise TaskGroupError(bad_tasks)
Jul 31 14:46:44 hook electrumx_server[4680]: aiorpcx.curio.TaskGroupError: TaskGroupError(BadPeerError)
Jul 31 14:46:44 hook electrumx_server[4680]: INFO:SessionManager:closing down RPC, TCP, SSL listening servers
Jul 31 14:46:44 hook electrumx_server[4680]: ERROR:TaskGroup:task crashed: <Task finished coro=<SessionManager.serve() done, defined at /opt/python/lib/python3.6/site-packages/electrumX-1.7.2-py3.6.egg/electrumx/server/session.py:398> exception=TaskGroupError([<Task finished coro=<PeerManager._monitor_peer() done, defined at /opt/python/lib/python3.6/site-packages/electrumX-1.7.2-py3.6.egg/electrumx/server/peers.py:185> exception=TaskGroupError([<Task finished coro=<PeerManager._send_headers_subscribe() done, defined at /opt/python/lib/python3.6/site-packages/electrumX-1.7.2-py3.6.egg/electrumx/server/peers.py:310> exception=BadPeerError('bad height 534,485 (ours: 534,540)',)>],)>],)>
Jul 31 14:46:44 hook electrumx_server[4680]: Traceback (most recent call last):
Jul 31 14:46:44 hook electrumx_server[4680]:   File "/opt/python/lib/python3.6/site-packages/aiorpcX-0.5.9-py3.6.egg/aiorpcx/curio.py", line 138, in _on_done
Jul 31 14:46:44 hook electrumx_server[4680]:     task.result()
Jul 31 14:46:44 hook electrumx_server[4680]:   File "/opt/python/lib/python3.6/site-packages/electrumX-1.7.2-py3.6.egg/electrumx/server/session.py", line 425, in serve
Jul 31 14:46:44 hook electrumx_server[4680]:     await group.spawn(self._housekeeping())
Jul 31 14:46:44 hook electrumx_server[4680]:   File "/opt/python/lib/python3.6/site-packages/aiorpcX-0.5.9-py3.6.egg/aiorpcx/curio.py", line 254, in __aexit__
Jul 31 14:46:44 hook electrumx_server[4680]:     await self.join(wait=self._wait)
Jul 31 14:46:44 hook electrumx_server[4680]:   File "/opt/python/lib/python3.6/site-packages/aiorpcX-0.5.9-py3.6.egg/aiorpcx/curio.py", line 228, in join
Jul 31 14:46:44 hook electrumx_server[4680]:     raise TaskGroupError(bad_tasks)
Jul 31 14:46:44 hook electrumx_server[4680]: aiorpcx.curio.TaskGroupError: TaskGroupError(TaskGroupError)
Jul 31 14:46:44 hook electrumx_server[4680]: INFO:BlockProcessor:flushing to DB for a clean shutdown...
Jul 31 14:46:44 hook electrumx_server[4680]: ERROR:TaskGroup:task crashed: <Task finished coro=<Controller.serve() done, defined at /opt/python/lib/python3.6/site-packages/electrumX-1.7.2-py3.6.egg/electrumx/server/controller.py:80> exception=TaskGroupError([<Task finished coro=<SessionManager.serve() done, defined at /opt/python/lib/python3.6/site-packages/electrumX-1.7.2-py3.6.egg/electrumx/server/session.py:398> exception=TaskGroupError([<Task finished coro=<PeerManager._monitor_peer() done, defined at /opt/python/lib/python3.6/site-packages/electrumX-1.7.2-py3.6.egg/electrumx/server/peers.py:185> exception=TaskGroupError([<Task finished coro=<PeerManager._send_headers_subscribe() done, defined at /opt/python/lib/python3.6/site-packages/electrumX-1.7.2-py3.6.egg/electrumx/server/peers.py:310> exception=BadPeerError('bad height 534,485 (ours: 534,540)',)>],)>],)>],)>
Jul 31 14:46:44 hook electrumx_server[4680]: Traceback (most recent call last):
Jul 31 14:46:44 hook electrumx_server[4680]:   File "/opt/python/lib/python3.6/site-packages/aiorpcX-0.5.9-py3.6.egg/aiorpcx/curio.py", line 138, in _on_done
Jul 31 14:46:44 hook electrumx_server[4680]:     task.result()
Jul 31 14:46:44 hook electrumx_server[4680]:   File "/opt/python/lib/python3.6/site-packages/electrumX-1.7.2-py3.6.egg/electrumx/server/controller.py", line 116, in serve
Jul 31 14:46:44 hook electrumx_server[4680]:     serve_externally_event.set()
Jul 31 14:46:44 hook electrumx_server[4680]:   File "/opt/python/lib/python3.6/site-packages/aiorpcX-0.5.9-py3.6.egg/aiorpcx/curio.py", line 254, in __aexit__
Jul 31 14:46:44 hook electrumx_server[4680]:     await self.join(wait=self._wait)
Jul 31 14:46:44 hook electrumx_server[4680]:   File "/opt/python/lib/python3.6/site-packages/aiorpcX-0.5.9-py3.6.egg/aiorpcx/curio.py", line 228, in join
Jul 31 14:46:44 hook electrumx_server[4680]:     raise TaskGroupError(bad_tasks)
Jul 31 14:46:44 hook electrumx_server[4680]: aiorpcx.curio.TaskGroupError: TaskGroupError(TaskGroupError)
```

**Comment by kyuupichan** [comment]
Almost certain this is fixed in git HEAD; I haven't done a tagged release with it yet

**Comment by SomberNight** [comment]
same as https://github.com/kyuupichan/electrumx/issues/538

**Comment by smmalis37** [comment]
Could we get a 1.7.3 with this fix in it released? I'm hitting this exception constantly now.

**Comment by kyuupichan** [comment]
I'll see if I can get something done today

**Comment by kyuupichan** [comment]
I've released 1.7.3
